sycl/test/extensions/WorkGroupMemory/api_misuse_test.cpp

```diff
@@ -0,0 +1,45 @@
+// RUN: %clangxx -fsycl -fsyntax-only -ferror-limit=30 -Xclang -verify -Xclang -verify-ignore-unexpected=note,warning %s
```

Comment:
We might as well remove the error-limit entirely. In the current version of the test it doesn't make much of a difference, but imagine a case where someone was to add additional cases to the test and either hits this limit exactly, making the test vulnerable to ignoring additional errors, or get confused when expected errors don't happen due to the limit.
```suggestion
// RUN: %clangxx -fsycl -fsyntax-only -ferror-limit=0 -Xclang -verify -Xclang -verify-ignore-unexpected=note,warning %s
```